HACKING.md

```diff
@@ -203,25 +203,29 @@ Coding Style
   *unless they avoid great repetition*. On the average case,
   just put the assignment on the previous line. So try NOT to do
   the following:
-  ~~```if ((ret = open(...)) == -1)```~~
+  ~~`if ((ret = open(...)) == -1)`~~
 * Explicit comparisons are better than implicit, i.e. prefer writing
-  ```if (number == 0)``` or ```if (pointer == NULL)```
-  instead of ~~```if (!number)```~~ or ~~```if (!pointer)```~~. It only makes
-  sense to test booleans directly, for example ```if (is_valid)``` is good.
+  `if (number == 0)` or `if (pointer == NULL)`
+  instead of ~~`if (!number)`~~ or ~~`if (!pointer)`~~. It only makes
+  sense to test booleans directly, for example `if (is_valid)` is good.
   Furthermore have the literal last in the comparison, not first, i.e.
-  prefer writing ```if (open(...) == -1)``` instead of
-  ~~```if (-1 == open(...))```~~.
+  prefer writing `if (open(...) == -1)` instead of
+  ~~`if (-1 == open(...))`~~.
 * Control statements need to have braces on separate line,
   no matter how simple they are.
   * Caution, do-while loops should have the closing brace at the same
     line with while, so that it can't be confused with empty while statement.
     ```c
     do
     {
-        /* ... */
+        // ...
 
     } while (condition);
     ```
+* In functions which can fail, error code (int) or success/failure (bool) should be returned.
+    * `true`(bool) and `0`(int) should always signify success.
+    * Only return an error code (int) when there are multiple different return values for different errors. If a function can only return `0` (success) or `-1` (error) use `bool` instead.
```

Comment:
Many times it's useful to combine error code and success return value in the same return number. I know I have coded several functions in the past that return `size_t` since it's about memory sizes, but in case of error they return `(size_t) -1`. Just keep that in mind while changing stuff.

Reply:
This is true, I might add a point about this. Return value like `recv()` where positive integers are valid and -1 means error is totally acceptable.